### PR TITLE
Create 0028-storybook-for-tdr-components-library.md

### PR DESCRIPTION
**Pull request by garethfoote**

### CODE REVIEW

File: architecture-decision-records/0028-storybook-for-tdr-components-library.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+>  (if you can't [use these functions to access the elements], it's possible your UI is inaccessible).
+
+The elements state or content is then asserted against expectations using Jest functions. These tests run in the UI allowing us to see the visual/interactive outcomes of these tests. Crucially, with Storybook, all of these tests will also run on the CLI during deployment. 
+
+Storybook also provides out-of-the-box accessibility testing using Axe Core, which would be a straight swap for existing setup. 
+
+The interaction tests are run using storybook addons, which are effectively wrappers for Jest and Testing Library. This means we are paritially 'buying into' the ecosystem of Storybook, however there is evidence of the library's maturity and widespread use. The alternative is to write more verbose testing code for sometimes quite simple components. The labour saved amongst other benefits means we are more likely to write robust and comprehensive tests covering the multiple states a component can exist in. 
+
```
Comment (TomJKing) [inline architecture-decision-records/0028-storybook-for-tdr-components-library.md:28]: Could you add a `Decision` section here. See other ADRs for examples. It should state that we decided to move to "StoryBook" based on the reasons stated in the considerations section.

File: architecture-decision-records/0028-storybook-for-tdr-components-library.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Storybook provides an out-of-box inteface that isolates each component UI state. You achieve this by writting 'Stories' in a Component Story Format (CSF). CSF is an open standard for UI component examples based on JavaScript ES6 modules. 
+
+Each story is a JS or TS file that contain exports, which configure variations of the UI. Each variation is rendered in the Storybook interface allowing for much more detailed QA and acceptance testing for design, dev and other team members. 
+
+Storybook provides a play function that 'is a small snippet of code that runs after a story finishes rendering. You can use this to test user workflows.' ([source](https://storybook.js.org/docs/react/writing-tests/interaction-testing)) Within the play function you can use a convenient [API from Testing-Library](https://testing-library.com/docs/queries/about) for querying the DOM. This library focusses on allow you to access only the visible items or those made available to the [accessiblity tree](https://developer.mozilla.org/en-US/docs/Glossary/Accessibility_tree). As it states in the docs for this API:
```
Comment (TomJKing) [inline architecture-decision-records/0028-storybook-for-tdr-components-library.md:19]: Re-word slightly: `... This library focusses on allowing you to access ...`

**Review by TomJKing — COMMENTED** [review]
Looks good. Thanks for writing this up. Just some minor comments.

**Comment by garethfoote** [comment]
All comments resolved @TomJKing

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**